### PR TITLE
Added headers to multimessages

### PR DESCRIPTION
**Pull request by dartartem**

### CODE REVIEW

File: eventuate-messaging-kafka-common/src/main/java/io/eventuate/messaging/kafka/common/EventuateKafkaMultiMessageConverter.java [hunk 1]
```diff
@@ -151,11 +224,24 @@ private int estimatedStringSizeInBytes(String s) {
...
+        MultiMessageEncoder.MessagesEncoder.HeadersEncoder messageHeadersEncoder =
+                messagesEncoder.headersCount(message.getHeaders().size());
+
+        message.getHeaders().forEach(header -> messageHeadersEncoder.next().key(header.getKey()).value(header.getValue()));
+
+        messagesEncoder.key(message.getKey()).value(message.getValue());
+      });
 
-      messagesToWrite.forEach(message -> messagesEncoder.next().key(message.getKey()).value(message.getValue()));
+      System.out.println(multiMessageEncoder.encodedLength() + messageHeaderEncoder.encodedLength());
```
Comment (cer) [inline eventuate-messaging-kafka-common/src/main/java/io/eventuate/messaging/kafka/common/EventuateKafkaMultiMessageConverter.java]: Please remove

File: eventuate-messaging-kafka-common/src/main/java/io/eventuate/messaging/kafka/common/EventuateKafkaMultiMessageConverter.java [hunk 2]
```diff
@@ -151,11 +224,24 @@ private int estimatedStringSizeInBytes(String s) {
...
+      headers.forEach(header -> headersEncoder.next().key(header.getKey()).value(header.getValue()));
+
       MultiMessageEncoder.MessagesEncoder messagesEncoder = multiMessageEncoder.messagesCount(messagesToWrite.size());
+      messagesToWrite.forEach(message -> {
+        messagesEncoder.next();
+
+        MultiMessageEncoder.MessagesEncoder.HeadersEncoder messageHeadersEncoder =
+                messagesEncoder.headersCount(message.getHeaders().size());
+
+        message.getHeaders().forEach(header -> messageHeadersEncoder.next().key(header.getKey()).value(header.getValue()));
```
Comment (cer) [inline eventuate-messaging-kafka-common/src/main/java/io/eventuate/messaging/kafka/common/EventuateKafkaMultiMessageConverter.java]: Would a for loop be more efficient, e.g. not cause heap allocation of lambda? if so, use it

File: eventuate-messaging-kafka-common/src/main/java/io/eventuate/messaging/kafka/common/EventuateKafkaMultiMessageConverter.java [hunk 3]
```diff
@@ -104,39 +166,50 @@ public MessageBuilder() {
...
-      return s == null ? 0 : s.length() * 2;
+    private boolean isSizeOverLimit(int estimatedSize) {
+      return maxSize.map(ms -> size + estimatedSize > ms).orElse(false);
     }
 
     public byte[] toBinaryArray() {
+      return toBinaryArray(false);
+    }
+
+    public byte[] toBinaryArray(boolean bufferAsIs) {
```
Comment (cer) [inline eventuate-messaging-kafka-common/src/main/java/io/eventuate/messaging/kafka/common/EventuateKafkaMultiMessageConverter.java]: What does `bufferAsIs` mean?

File: eventuate-messaging-kafka-common/src/main/java/io/eventuate/messaging/kafka/common/EventuateKafkaMultiMessageConverter.java (hunk 1 above)
Comment (dartartem) [reply eventuate-messaging-kafka-common/src/main/java/io/eventuate/messaging/kafka/common/EventuateKafkaMultiMessageConverter.java]: Sorry, I forgot to do it. Will do.

File: eventuate-messaging-kafka-common/src/main/java/io/eventuate/messaging/kafka/common/EventuateKafkaMultiMessageConverter.java (hunk 2 above)
Comment (dartartem) [reply eventuate-messaging-kafka-common/src/main/java/io/eventuate/messaging/kafka/common/EventuateKafkaMultiMessageConverter.java]: it should be

File: eventuate-messaging-kafka-common/src/test/java/io/eventuate/messaging/kafka/common/EventuateKafkaMultiMessageConverterTest.java [hunk 4]
```diff
@@ -11,152 +11,121 @@
...
+    Assert.assertTrue(messageBuilder.addMessage(SIMPLE_MESSAGES.get(0)));
+    Assert.assertFalse(messageBuilder.addMessage(SIMPLE_MESSAGES.get(1)));
   }
 
   @Test
   public void testMessageBuilderHeaderSizeCheck() {
     int sizeOfFirstMessage = 2 * 4
-            + MESSAGES.get(0).getKey().length() * 2
-            + MESSAGES.get(0).getValue().length() * 2;
+            + SIMPLE_MESSAGES.get(0).getKey().length() * 2
```
Comment (cer) [inline eventuate-messaging-kafka-common/src/test/java/io/eventuate/messaging/kafka/common/EventuateKafkaMultiMessageConverterTest.java]: The code would be simpler if there was a MESSAGES_0_KEY, MESSAGES_1_VALUE, ...?

File: eventuate-messaging-kafka-common/src/main/java/io/eventuate/messaging/kafka/common/EventuateKafkaMultiMessageConverter.java (hunk 3 above)
Comment (dartartem) [reply eventuate-messaging-kafka-common/src/main/java/io/eventuate/messaging/kafka/common/EventuateKafkaMultiMessageConverter.java]: I removed it by next commit.
But forgot to push.
Fixed it.
Used it to make result array length == encoded message size (not estimated)

File: eventuate-messaging-kafka-common/src/test/java/io/eventuate/messaging/kafka/common/EventuateKafkaMultiMessageConverterTest.java (hunk 4 above)
Comment (dartartem) [reply eventuate-messaging-kafka-common/src/test/java/io/eventuate/messaging/kafka/common/EventuateKafkaMultiMessageConverterTest.java]: ok, will do

**Pull request merged**